### PR TITLE
1049: pulled newest changes from develop

### LINKED ISSUE #1049: Work Package Creation - Make blocked by a multiple autocomplete
### Description

Make the blocked by text field in the work package creation be a mulitple autocomplete where the options are only other work packages in the same project. 

### Acceptance Criteria

Successfully implements a multiple autocomplete with the correct options. 

### Proposed Solution

See Task list modal for an example of a multiple auto complete, or member editing view on the teams page. You will want to replace the blocked by text field in the work package create form to be a multiple autocomplete. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by tigerhippo**
## Changes

Pulled up to date code from develop and added my changes back in

## Test Cases

FinishLine starts up + yarn test:frontend and backend

## To Do

Manually test on FinishLine

## Checklist

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1049

### CODE REVIEW

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 1]
```diff
@@ -74,6 +77,8 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
     startDate.setDate(startDate.getDate() + daysUntilNextMonday);
   }
   const query = useQuery();
+  const { data: project } = useSingleProject(validateWBS(wbsNum));
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:80]: add isLoading, isError and error and then validate all of that.

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx [hunk 2]
```diff
@@ -21,7 +21,8 @@ export interface CreateWorkPackageFormInputs {
   duration: number | null;
   crId: number;
   stage: WorkPackageStage | 'None';
-  blockedBy: { wbsNum: string }[];
+  wbsNum: string | null;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx:24]: why did u add this

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx [hunk 3]
```diff
@@ -52,8 +53,8 @@ const CreateWorkPackageForm: React.FC = () => {
         return;
       }
 
-      const blockedByWbsNums = blockedBy.map((blocker: { wbsNum: string }) => {
-        const blockedWbsNum = validateWBS(blocker.wbsNum);
+      const blockedByWbsNums = blockedBy.map((blocker: string) => {
+        const blockedWbsNum = validateWBS(blocker);
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx:57]: you can replace with with just validateWBS, so like blockedBy.map(validateWBS)

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 4]
```diff
@@ -86,13 +91,16 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
       crId: Number(query.get('crId')),
       stage: 'NONE' as WorkPackageStage | 'None',
       startDate,
+      wbsNum: wbsNum,
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:94]: why is this here

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 5]
```diff
@@ -74,6 +77,8 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
     startDate.setDate(startDate.getDate() + daysUntilNextMonday);
   }
   const query = useQuery();
+  const { data: project } = useSingleProject(validateWBS(wbsNum));
+  const workPackages = project ? project.workPackages : [];
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:81]: check if !project return <LoadingIndicator/> plus check if isError error page and isLoading loading indicator

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looks Great! just a couple cleanup things, I think you forgot to delete the wbsNum from the form, once thats finished it should be good

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 6]
```diff
@@ -119,25 +126,30 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
...
-      >
-        + ADD NEW BLOCKER
-      </Button>
+      <Controller
+        name="blockedBy"
+        control={control}
+        render={({ field: { onChange, value: formValue } }) => (
+          <Autocomplete
+            isOptionEqualToValue={(option, value) => {
+              console.log(option, value);
```
Comment (anthonybernardi) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:135]: delete (and then the function can just be `(option, value) => option.id === value.id`

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 7]
```diff
@@ -119,25 +126,30 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
...
+              return option.id === value.id;
+            }}
+            filterSelectedOptions
+            multiple
+            options={blockedByOptions}
+            getOptionLabel={(option) => option.label}
+            onChange={(_, values) => onChange(values.map((v) => v.id))}
+            value={formValue.map((v: string) => {
+              let change = blockedByOptions.find((o) => o.id === v);
```
Comment (anthonybernardi) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:144]: const

**Review by anthonybernardi — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 8]
```diff
@@ -103,41 +107,55 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
...
+        render={({ field: { onChange, value: formValue } }) => (
+          <Autocomplete
+            isOptionEqualToValue={(option, value) => option.id === value.id}
+            filterSelectedOptions
+            multiple
+            options={blockedByOptions}
+            getOptionLabel={(option) => option.label}
+            onChange={(_, values) => onChange(values.map((v) => v.id))}
+            value={formValue.map((v: string) => {
+              const change = blockedByOptions.find((o) => o.id === v);
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:150]: rename these variable from o and v, the name's or so vague that it's confusing

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 9]
```diff
@@ -103,41 +107,55 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
...
 
+  if (useSingleProjectIsError) return <ErrorPage message={useSingleProjectError?.message} />;
+  if (!project || isProjectLoading) return <LoadingIndicator />;
+  const workPackages = project ? project.workPackages : [];
+
+  const blockedByOptions = workPackages.map(blockedByToAutocompleteOption);
+
   const blockedByFormControl = (
     <FormControl fullWidth>
-      <FormLabel>Blocked By</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:122]: add this back in, looks weird without it

**Review by RChandler234 — CHANGES_REQUESTED** [review]
when you have work packages selected, and then change the project, it crashes the frontend rn, I think you'll need to somehow clear the selected blocked bys whenever the project is changed.

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 10]
```diff
@@ -103,41 +108,56 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
...
+        render={({ field: { onChange, value: formValue } }) => (
+          <Autocomplete
+            isOptionEqualToValue={(option, value) => option.id === value.id}
+            filterSelectedOptions
+            multiple
+            options={blockedByOptions}
+            getOptionLabel={(option) => option.label}
+            onChange={(_, values) => onChange(values.map((value) => value.id))}
+            value={formValue.map((value: string) => {
+              const change = blockedByOptions.find((option) => option.id === value);
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:152]: quick thing just delete the change thing to just be blockedByOptions.find...! and then delete the curly braces

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 11]
```diff
@@ -147,9 +167,8 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({
   ) => {
     if (value) {
       setWbsNum(value.id);
-    } else {
-      setWbsNum('');
     }
+    setValue('blockedBy', [] as string[]);
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:171]: dont you have to put this before setting the wbsNum

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**